cpp/include/cuspatial/experimental/detail/point_in_polygon.cuh

```diff
@@ -72,22 +73,36 @@ __device__ inline bool is_point_in_polygon(Cart2d const& test_point,
     // for each line segment, including the segment between the last and first vertex
     for (auto point_idx = ring_begin; point_idx < ring_end; point_idx++) {
       Cart2d const a = poly_points_first[point_idx];
-      y1_flag        = a.y > test_point.y;
-      if (y1_flag != y0_flag) {
-        T run           = b.x - a.x;
-        T rise          = b.y - a.y;
-        T rise_to_point = test_point.y - a.y;
+      T run          = b.x - a.x;
+      T rise         = b.y - a.y;
+
+      // Points on the line segment are the same, so intersection is impossible.
+      // This is possible because we allow closed or unclosed polygons.
+      if (run == 0.0 && rise == 0.0) continue;
```

Comment:
Can you try to replace this compare with the new utility I added in https://github.com/rapidsai/cuspatial/pull/773?
I'd use as `if (float_eq_by_ulp(b.x, a.x) && float_eq_by_ulp(b.y, a.y)) continue;`

Reply:
(No need to push to this PR for now - just want to test how usable that utility is)

Reply:
I was able to test it. Looks like it improves our accuracy for detecting those really hard corner cases like @harrism had identified.

```
            [Point([3.33, 1.11]), Polygon([[6, 2], [3, 1], [3, 4], [6, 2]]), True],
        ],
    )
    def test_float_precision_limits(point, polygon, expects):
        gpdpoint = gpd.GeoSeries(point)
        gpdpolygon = gpd.GeoSeries(polygon)
        point = cuspatial.from_geopandas(gpdpoint)
        polygon = cuspatial.from_geopandas(gpdpolygon)
        result = polygon.contains(point)
>       assert gpdpolygon.contains(gpdpoint).values == result.values_host
```

Produces a false now, previously True. It is good for us to have improved accuracy but in this case we conflict with Pandas, who produces True.

Reply:
Isn't this because GeoPandas `contains` uses Shapely `contains` and not `contains_properly`? By excluding boundary points, you are now implementing `contains_properly`, not `contains`. https://shapely.readthedocs.io/en/latest/reference/shapely.contains_properly.html